src/discord_utils.py

```diff
@@ -31,6 +31,27 @@ async def try_set_pin(message: Message, pin_state: bool) -> None:
         print("Altering message pin state failed:", e)
 
 
+def build_filepath_for_attachment(message_id: int, attachment: Attachment) -> str:
+    """Generate a unique filepath for a given attachment for the purposes of saving it to disk."""
```

Comment:
Did you mean to only include the first line of the descriptions in https://github.com/anoadragon453/busty/pull/167#discussion_r1103884933 and https://github.com/anoadragon453/busty/pull/167#discussion_r1103887258?

Reply:
Yeah, I agreed with Jack on specifying the exact format is unnecessary and just takes up space.
https://github.com/anoadragon453/busty/pull/167#discussion_r1106192333

Reply:
Yeah, I agreed with Jack on specifying the exact format of this function was unnecessary and just takes up space, so I've left it to describe what the function does when given an attachment and message id.
https://github.com/anoadragon453/busty/pull/167#discussion_r1106192333

Reply:
Hum, I still think useful information from those comments is being left out, but I agree that at least the descriptions of the arguments and return types are sometimes redundant. In that case, we can leave `build_filepath_for_attachment` alone.

For `build_filepath_for_attachment`, it's useful for callers to know that the returned filepath will be absolute and located in the attachment dir, so we should at least add that bit?

```suggestion
    """Generate a unique, absolute filepath for a given attachment located in the configured attachment directory."""
```

As for the description of the makeup of the filepath, we should still describe *why* we chose that as the format. Doing so helped us understand that our logic was in fact flawed, and we didn't need to include the Discord message ID in order to get a unique filepath.

That all may be better left to a comment in the method though (i.e. below), as we know calling functions in our codebase won't parse this filepath. If this were a public function in a library, I'd be more inclined to include the below in the docstring, for the caller may end up needing to know the structure of the returned path (in order to extract part of it).

```py
	# Generate and return a filepath in the following format:
    #     <attachment_directory>/<Discord message ID>.<attachment ID>.<file extension>
    # For example:
    #     /home/user/busty/625891304148303894.625891304081063986.mp3
    # 
    # The Discord Message ID is included as well, as individual messages can have multiple attachments.
    filepath = path.join(
        config.attachment_directory_filepath,
        "{}.{}{}".format(
            message_id,
            attachment.id,
            os.path.splitext(attachment.filename)[1],
        ),
    )
    return filepath
```

In general I tend to over-document rather than under-document. It can become especially useful when your codebase becomes larger and older, and you're left having to crawl through PRs or message people in order to understand why certain choices were made and whether changing code will break something.